flink-dist/src/main/flink-bin/bin/flink

```diff
@@ -17,20 +17,41 @@
 # limitations under the License.
 ################################################################################
 
-target="$0"
 # For the case, the executable has been directly symlinked, figure out
 # the correct bin path by following its symlink up to an upper bound.
 # Note: we can't use the readlink utility here if we want to be POSIX
 # compatible.
-iteration=0
-while [ -L "$target" ]; do
-    if [ "$iteration" -gt 100 ]; then
-        echo "Cannot resolve path: You have a cyclic symlink in $target."
+followSymLink() {
+    local iteration=0
+    local bar=$1
+    while [ -L "$bar" ]; do
+        if [ "$iteration" -gt 100 ]; then
+            echo "Cannot resolve path: You have a cyclic symlink in $bar."
+            break
+        fi
+        ls=`ls -ld -- "$bar"`
+        bar=`expr "$ls" : '.* -> \(.*\)$'`
+        iteration=$((iteration + 1))
+    done
+
+    echo "$bar"
+}
+
+target=$(followSymLink "$0")
+
+#Search --configDir into the parameters and set it as FLINK_CONF_DIR
+args=("$@")
+for i in "${!args[@]}"; do
+    if [ ${args[$i]} = "--configDir" ]; then
+        dir=$(followSymLink "${args[$(($i+1))]}" )
+        if [ -d "$dir" ]; then
+            FLINK_CONF_DIR=`cd "${dir}"; pwd -P`
```

Comment:
Added chained version, the second command runs only if the first was successful; i think it is a more correct behaviour.